### ISSUE TITLE
aws_ec2 module only looks at first item in hostnames list

### ISSUE THREAD

**Issue by fostermi** [ansible-collections/amazon.aws#583]
### Summary

The section for defining how `hostnames` are reported in the inventory shows it as a YAML list:
```
hostnames:
  - tag:Name=Tag1,Name=Tag2  # Return specific hosts only
  - tag:CustomDNSName
  - dns-name
  - name: 'tag:Name=Tag1,Name=Tag2'
  - name: 'private-ip-address'
    separator: '_'
    prefix: 'tag:Name'
```

However, creating an example aws_ec2.yml file such as the following:
```
---
plugin: aws_ec2
regions:
  - us-east-1
keyed_groups:
  - prefix: tag
    key: tags
hostnames:
   - tag:Tag1
   - tag:Tag2 
compose:
  # Use the private IP address to connect to the host
  # (note: this does not modify inventory_hostname, which is set via I(hostnames))
  ansible_host: private_ip_address
```

 the module only uses the first item in the list instead of all items. Hence only the single host with the tag "Tag1" is returned.

Additionally, the module prepends the name of tag Key to the hostname, instead of just the value of the key.

### Issue Type

Bug Report

### Component Name

aws_ec2

### Ansible Version

```console (paste below)
$ ansible --version
ansible 2.9.10
```

### Collection Versions

```console (paste below)
$ ansible-galaxy collection list
```

### AWS SDK versions

```console (paste below)
$ pip show boto boto3 botocore
```
```
Name: boto3
Version: 1.19.10
Summary: The AWS SDK for Python
Home-page: https://github.com/boto/boto3
Author: Amazon Web Services
Author-email: 
License: Apache License 2.0
Location: /Users/mifo4973/venvs/ansible-2.9.10-aws/lib/python3.7/site-packages
Requires: botocore, jmespath, s3transfer
Required-by: 
---
Name: botocore
Version: 1.22.10
Summary: Low-level, data-driven core of boto 3.
Home-page: https://github.com/boto/botocore
Author: Amazon Web Services
Author-email: 
License: Apache License 2.0
Location: /Users/mifo4973/venvs/ansible-2.9.10-aws/lib/python3.7/site-packages
Requires: jmespath, python-dateutil, urllib3
Required-by: awscli, boto3, s3transfer
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### Steps to Reproduce

```yaml (paste below)
---
plugin: aws_ec2
regions:
  - us-east-1
keyed_groups:
  - prefix: tag
    key: tags
hostnames:
   - tag:Tag1
   - tag:Tag2 
compose:
  # Use the private IP address to connect to the host
  # (note: this does not modify inventory_hostname, which is set via I(hostnames))
  ansible_host: private_ip_address
```

### Expected Results

I expect that all the tags in the hostnames list will be returned as an ansible_host.

```
"hostvars": {
            "Test1": {
....

"hostvars": {
            "Test2": {
```

### Actual Results

```console (paste below)
{
    "_meta": {
        "hostvars": {
            "Tag1_Test1": {
...
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by alinabuzachis** [comment]
@fostermi Could you please give us an update on this issue?

**Comment by ansibullbot** [comment]
@fostermi This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by alinabuzachis** [comment]
@fostermi I'm going to close this issue. Please give a try to this patch https://github.com/ansible-collections/amazon.aws/pull/862 and feel free to open a new issue if you are still facing the issue. Thanks.
